provisioning_client/tests/hsm_client_riot_ut/hsm_client_riot_ut.c

```diff
@@ -175,17 +196,24 @@ static char* umocktypes_stringify_RIOT_ECC_PUBLIC(const RIOT_ECC_PUBLIC* value)
     }
     else
     {
-        int length = snprintf(NULL, 0, "{ %p, %p, %d }",
-            value->x.data, value->y.data, (int)value->infinity);
+        int length = snprintf(NULL, 0, "{ %d, %zd, %p }",
+            value->X.s, value->X.n, value->X.p);
+        length += snprintf(NULL, 0, "{ %d, %zd, %p }",
+            value->Y.s, value->Y.n, value->Y.p);
+        length += snprintf(NULL, 0, "{ %d, %zd, %p }",
+            value->Z.s, value->Z.n, value->Z.p);
```

Comment:
Take the following sequence of `snprintf` calls:
1) Returns `20`
2) Returns `-1`
3) Returns `-1`

The if check below will compare `18 < 0` and it will proceed to malloc space for 18 and try to copy and fail. Should be updated to account for that case.

Reply:
Fixed: merge into 1 snprintf call